Client/Frontend/Login Management/BreachAlertsManager.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  BreachAlertsManager.swift
+//  Client
+//
+//  Created by Vanna Phong on 5/21/20.
+//  Copyright © 2020 Mozilla. All rights reserved.
+//
+
+import Foundation
+import Storage // or whichever module has the LoginsRecord class
+
+/// Breach structure decoded from JSON
+struct BreachRecord: Codable {
+    var name: String
+    var title: String
+    var domain: String
+    var breachDate: String
+    var addedDate: String
+    var modifiedDate: String
+    var pwnCount: Int
+    var description: String
+    var logoPath: String
+    var isVerified: Bool
+    var isFabricated: Bool
+    var isSensitive: Bool
+    var isRetired: Bool
+    var isSpamList: Bool
+    var logoUrl: String?
+}
+
+/// A manager for the user's breached login information, if any.
+public class BreachAlertsManager {
+
+    //
+    // MARK: - Variables and Properties
+    //
+    var dataTask: URLSessionDataTask?
+    var endpointURL = "https://monitor.firefox.com/hibp/breaches"
+    var breaches: [BreachRecord] = []
+
+    //
+    // MARK: - Internal Methods
+    //
+    /// Loads breaches from Monitor endpoint.
+    public func loadBreaches() {
+        guard let url = URL(string: endpointURL) else {
+            return
+        }
+
+        dataTask?.cancel()
+
+        dataTask = URLSession.shared.dataTask(with: url) { data, response, error in
+            guard self.validatedHTTPResponse(response) != nil else {
+                print("loadBreaches(): invalid HTTP response")
+                return
+            }
+
+            if let error = error {
+                print("loadBreaches(): error: \(error)")
+                return
+            }
+
+            guard let data = data, !data.isEmpty else {
+                print("loadBreaches(): invalid data")
+                return
+            }
+
+            let decoder = JSONDecoder()
+
+            // .convertFromPascalCase
+            decoder.keyDecodingStrategy = .custom { keys in
+                let key = keys.last! // make sure array not empty
+
+                // string manipulation
+                let keyStr = key.stringValue
+                let pascalToCamel = keyStr.prefix(1).lowercased() + keyStr.dropFirst()
+
+                // CodingKey conformity
+                let codingKeyType = type(of: key)
+                return codingKeyType.init(stringValue: pascalToCamel)!
+
+            }
+
+            if let decoded = try? decoder.decode([BreachRecord].self, from: data) {
+                DispatchQueue.main.async {
+                    self.breaches = decoded
+                }
+            }
+        }
+
+        dataTask?.resume()
+
+    }
```

Comment:
I'm wondering if we should separate the actual data request out into a client that just handles doing that work? Call it BreachAlertsClient or something like that? Looking around the app I see that as a pattern for data tasks.

Reply:
When I study the cases with the other clients used in the app, they either seem to be 1) whatever is calling the client is complex or 2) the network requests are many/complex. I think in this case, keeping this network call just confined to this class should be okay, since it's pretty simple and the max it will do is change the UI.

Reply:
Chatted outside of github about this and we came to the conclusion that a Client will be added for better testability and future expansion of breach alerts requests, if ever necessary.